x/evmutil/keeper/erc20.go

```diff
@@ -0,0 +1,144 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/kava-labs/kava/x/evmutil/types"
+)
+
+const (
+	erc20BalanceOfMethod = "balanceOf"
+)
+
+// DeployMintableERC20Contract deploys an ERC20 contract on the EVM as the
+// module account and returns the address of the contract. This contract has
+// minting permissions for the module account.
+// Derived from tharsis/evmos
+// https://github.com/tharsis/evmos/blob/ee54f496551df937915ff6f74a94732a35abc505/x/erc20/keeper/evm.go
+func (k Keeper) DeployMintableERC20Contract(
+	ctx sdk.Context,
+	name string,
+	symbol string,
+	decimals uint8,
+) (types.InternalEVMAddress, error) {
```

Comment:
nit: If this method is only used in tests for now I'd maybe make it a bit more obvious (like put "Test" in the name or move it off the main Keeper). Want to avoid lower quality test code accidentaly making it into production in a future PR. Somewhat moot as this code is already on mainnet, but we might want to review the contract code before using it in this module.